flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManager.java

```diff
@@ -189,15 +180,17 @@ public boolean stopWorker(final KubernetesWorkerNode worker) {
 	public void onAdded(List<KubernetesPod> pods) {
 		runAsync(() -> {
 			for (KubernetesPod pod : pods) {
-				if (numPendingPodRequests > 0) {
-					numPendingPodRequests--;
+				WorkerResourceSpec workerResourceSpec = podWorkerResources.get(pod.getName());
+				final int pendingNum = getNumPendingWorkersFor(workerResourceSpec);
+				if (pendingNum > 0) {
```

Comment:
What happens if `pendingNum <= 0`? Shouldn't we return/stop the pod? This is actually out of the scope for this PR but it still looks not right.

Reply:
I double checked with @wangyang0918  about on what conditions will `onAdded` be called. There are two possible reasons that `onAdded` is called while there's no pending worker request.
- A duplicated `onAdded` message is received. In such cases, `workerNodes` should already contains the pod name. We can check that and simply ignore the duplicated message.
- `onAdded` message for a pod requested from the previous attempt. It is guaranteed that such pods is already added to `workerNodes` in `recoverWorkerNodesFromPreviousAttempts` before the `onAdded` is called, so we can apply the same check and ignore the duplicated pods.

Reply:
Recently i also realize that the `onAdded` for a same `Pod` may be called more that once here, e.g. some network problems cause the pod watcher reconnect, so it will be good to check the existence first. Otherwise, the `numPendingPodRequests` may be decreased wrongly.